src/docs/arch-hs.dox

```diff
@@ -4,5 +4,111 @@ The History Store in WiredTiger tracks old (all but the latest committed)
+with a globally visible tombstone. When this flag is set, it suppresses the effect of other
+visibility flags.
+
+- When the flag \c WT_CURSTD_HS_READ_COMMITTED is set, cursor interface can return any record except
+for records that have a globally visible tombstone. This flag has no effect when the flag
+\c WT_CURSTD_HS_READ_ALL is set.
+
+- When none of the flags discussed above are set on the cursor, snapshot based visibility checks are
+performed on all records returned to the API user. History Store cursor users must use one of
+the flags discussed above if the caller thread is running in a lower isolation level and
+doesn't hold a valid snapshot.
+
+@section arch_hs_reconciliation Historty Store and reconciliation
+When a modified user page is processed by either checkpoint or eviction module, the reconciliation
+code examines the update chain for each key and decides on the latest committed update that will be
+written as the on-disk value for the key. All the updates that are older than the on-disk version
+but not yet globally visible are added to the History Store table. The reconciliation code also
+fixes out-of-order timestamps for updates being added to the History Store table.
+
+When a modified History Store table page is processed by reconciliation, certain records on the page
+are removed when creating a new page image if the tombstones on those records have become gobally
+visible. This ensures that WiredTiger is only keeping the relevant historical data required to
+serve the oldest reader in the system or as dictated by the oldest timestamp set by the application.
+
+@section arch_hs_read Searching for older versions of a key in Historty Store
+When looking for an update visible to the current trasnaction, WiredTiger first searches the update
+chain for any visible updates. If the there is no visible update in the chain, WiredTiger then
+insepcts the on-disk version of the key. If that version is not visible to the transaction
+as well, WiredTiger searches the History Store table for a visible update for the key. When
+searching for updates that satisfy the read timestamp constraint, WiredTiger starts with the  newest
+update of the key in the History Store table and then iterates through the older updates until there
+are no updates left to process. Note that although there can be multiple updates in the
+History Store for a key and read timestamp combination, each update would have different
+visibility based on transaction based visibility rules.
+
+@section arch_hs_rts Historty Store and Rollback-to-stable
+Rollback-to-stable searches the History Store table for valid updates that are stable according to
+the supplied stable timestamp and replaces the on-disk version with an update from the History Store
+table that fulfills the criteria. Rollback-to-stable also removes updates from the
+History Store table that have a start timestamp newer than the stable timestamp.
```

Comment:
Now it also rollback and removes updates based on transaction ids.

Reply:
Fixed.